show-geodesic-path-between-two-points/src/main/java/com/esri/arcgismaps/sample/showgeodesicpathbetweentwopoints/MainActivity.kt

```diff
@@ -0,0 +1,193 @@
+        }
+    }
+
+    // marker graphic for the destination
+    private val endLocationMarkerGraphic by lazy {
+        Graphic().apply {
+            symbol = locationMarkerSymbol
+        }
+    }
+
+    // the geodesic path represented as line graphic
+    private val geodesicPathGraphic by lazy {
+        val lineSymbol = SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.red, 5f)
+        Graphic().apply {
+            symbol = lineSymbol
+        }
+    }
+
+    // the unit of distance measurement in kilometers
+    private val unitsOfMeasurement = LinearUnit(LinearUnitId.Kilometers)
+
+
+    // starting location for the distance calculation
+    private val startingPoint = Point(-73.7781, 40.6413, SpatialReference.wgs84())
+
+
+    // creates a graphic overlay to draw all graphics
+    private val graphicsOverlay = GraphicsOverlay()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        // authentication with an API key or named user is
+        // required to access basemaps and other location services
+        ArcGISEnvironment.apiKey = ApiKey.create(BuildConfig.API_KEY)
+        lifecycle.addObserver(mapView)
+
+        // create and add a map with a navigation night basemap style
+        val map = ArcGISMap(BasemapStyle.ArcGISImageryStandard).apply {
+            initialViewpoint = Viewpoint(Point(34.77, -10.24), 20e7)
+        }
+        // configure mapView assignments
+        mapView.apply {
+            this.map = map
+            // add the graphics overlay to the mapview
+            graphicsOverlays.add(graphicsOverlay)
+        }
+
+        // add all our marker graphics to the graphics overlay
+        graphicsOverlay.graphics.addAll(
+            listOf(startingLocationMarkerGraphic, endLocationMarkerGraphic, geodesicPathGraphic)
+        )
+
+        lifecycleScope.launch {
+            // check if the map has loaded successfully
+            map.load().onSuccess {
+                // capture and collect when the user taps on the screen
+                mapView.onSingleTapConfirmed.collect { event ->
+                    event.mapPoint?.let { point -> displayGeodesicPath(point) }
+                }
+            }.onFailure {
+                // if map load failed, show the error
+                showError("Error Loading Map")
+            }
+        }
+    }
+
+    /**
+     * Displays the destination location marker at the tapped location
+     * and draws a geodesic path curve using GeometryEngine.densifyGeodetic
+     * and computes the distance using GeometryEngine.lengthGeodetic
+     */
+    private fun displayGeodesicPath(point: Point) {
+        // project the tapped point into the same spatial reference as source point
+        val destinationPoint = GeometryEngine.project(point, SpatialReference.wgs84())
+
+        // check if the destination point is within the map bounds
+        // isEmpty returns true if out of bounds
```

Comment:
I tested the sample and it does seem to be possible to place the destination outside the map bounds. Not a big deal IMO but if that behaviour is supported then this check & comment should be removed.

Reply:
I noticed the projection does return valid coordinates when outside the map bounds. The bounds seem to be at +85,-85 lat. Anything outside +90,-90 lat is returned as NaN, NaN for which isEmpty() returns true